### ISSUE TITLE
Renaming and replacing at same time doesn't update fileName

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#1396]
**Describe the bug**
After replacing a file, if you update the file name in the table row, then click Submit, the file name does not get updated in the system metadata.

**To Reproduce**

Open dataset in editor.
Replace file.
Type a new name for the file in the DataItemView
Click Submit
Click View dataset, and see that the old file name is still there.

See screengrab:
![replace](https://user-images.githubusercontent.com/6643222/82463305-49096580-9a82-11ea-844b-3379ba200002.gif)

**Expected behavior**
The fileName should be updated in the system metadata.

**Desktop (please complete the following information):**
 - OS: Mac OS 10.15.4
 - Browser: Chrome 81.0.4044.122

**Additional context**
I've merged the releases branch into `i653-replace-item`, so make sure you `pull` before reproducing :)

**Comment by amoeba** [comment]
Hrm. I'm actually seeing something weirder. Replacing an item then renaming it causes the item to not be replaced at all. I did find one bug unrelated to the replace feature that might be related but doesn't explain what I'm seeing. I'll keep looking.

**Comment by amoeba** [comment]
Okay, this is now fixed.

This was a new bug introduced with the replace functionality due to how I had to fiddle with the object being replaced to get the editor to send the right API calls. See the commit (above) and its message for more detail. 

It strikes me as a bit messy for something that feels like it should be simple but it works. Anywhere in the UI that deals with a replaced item's sysmeta will need to behave this way.
